src/filesystem/mod.rs

```diff
@@ -0,0 +1,10 @@
+/// Error
+mod error;
+/// Describes how to read and write fs metadata
+pub mod metadata;
```

Comment:
How many of these do we need to be pub / can you get away with just pub use metadata::FsMetadata

Reply:
we can get away with far fewer pubs throughout this project, but IMO that is a separate piece of tech debt to determine what should or shouldn't be included in our lib